src/main/protobuf/ucfg.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+option java_package = "org.sonar.ucfg.protobuf";
+option optimize_for = SPEED;
+
+message UCFG {
+    string methodId = 1;
+    Location location = 2;
+    repeated string parameters = 3;
+    repeated string ucfgEntries = 4;
+    repeated BasicBlock basicBlocks = 5;
+}
+
+message BasicBlock {
+    string id = 1;
+    Location location = 2;
+    repeated Instruction instruction = 3;
+    oneof terminator {
+        Return ret = 4;
+        Jump jump = 5;
+    }
+}
+
+message Instruction {
+    Location location = 1;
+    string lhs = 2;
+    string methodId = 3;
+    repeated Expression args = 4;
+}
+
+message Return {
+    Location location = 1;
+    Expression returnedExpression = 2;
+}
+
+message Jump {
+    repeated string destinations = 1;
+}
+
+message Expression {
+    oneof expr {
```

Comment:
@benzonico not very elegant to have this meaningless nesting but probably there is no other way if want to avoid duplication

Reply:
Trade off between repeat and extensibility. I am pretty happy with the choice to say that expression today is one of two. Might be more in the future and I don't want to modify all the messages using an expression.